Makefile

```diff
@@ -0,0 +1,56 @@
+#   Copyright The beehive Authors.
+
+WHALE = "🇧 "
+OMG = "😱"
+
+PKG=github.com/kubeedge/beehive
+
+# Project packages.
+PACKAGES=$(shell go list ./... | grep -v "/vendor/")
+INTEGRATION_PACKAGE=${PKG}
+
+#Replaces ":" (*nix), ";" (windows) with newline for easy parsing
+GOPATHS=$(shell echo ${GOPATH} | tr ":" "\n" | tr ";" "\n")
+
+# Flags passed to `go test`
+TESTFLAGS ?= -v -race
+
+.PHONY: all test lint
+
+.DEFAULT: default
+
+all: test lint ## default
+	@echo "$(WHALE) $@"
+
+lint: ## golangci-lint check
+	@echo "$(WHALE) $@"
+	golangci-lint run --disable-all -E gofmt -E golint ./...
+	go vet ./...
+
+test: ## test case
+	@echo "$(WHALE) $@"
+	@go test ${TESTFLAGS} $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES})
+
+# https://deepzz.com/post/study-golang-test.html
+# https://deepzz.com/post/the-command-flag-of-go-test.html
+benchmark: ## run benchmarks tests
+	@echo "$(WHALE) $@"
+	@go test ${TESTFLAGS} $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES})  -bench . -run Benchmark 
+	
+coverage: ## generate coverprofiles from the unit tests, except tests that require root
+	@echo "$(WHALE) $@"
+	@rm -f coverage.txt
+	@go test -i ${TESTFLAGS} $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES}) 2> /dev/null
+	@( for pkg in $(filter-out ${INTEGRATION_PACKAGE},${PACKAGES}); do \
+		go test ${TESTFLAGS} \
+			-cover \
+			-coverprofile=profile.out \
+			-covermode=atomic $$pkg || exit; \
+		if [ -f profile.out ]; then \
+			cat profile.out >> coverage.txt; \
+			rm profile.out; \
+		fi; \
+	done )
+
+help: ## this help
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
```

Comment:
I use GNU Make 4.2.1 and I didn't have the variable MAKEFILE_LIST in it. So the help target will do nothing.